src/main/java/bio/terra/app/usermetrics/BardClient.java

```diff
@@ -23,30 +29,46 @@ public class BardClient {
 
   private final UserMetricsConfiguration metricsConfig;
 
-  private final RestTemplate restTemplate;
+  private final RestTemplate apiRestTemplate;
+
+  private final RestTemplate syncRestTemplate;
   private final HttpHeaders headers;
 
+  private static final int BEARER_TOKEN_CACHE_TIMEOUT_SECONDS = 3600;
+
+  private final Map<String, Object> bearerCache;
+
   @Autowired
   public BardClient(UserMetricsConfiguration metricsConfig) {
-    this.restTemplate = new RestTemplate();
-    restTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+    this.apiRestTemplate = new RestTemplate();
```

Comment:
I'm curious: does it need a whole new `RestTemplate`?  Seems like overkill to have one for each call no?

Reply:
This was surprising to me as well.  I would periodically see flaky test results without this change.

Reply:
JDK 11 added `HttpClient`, which does a lot of what `RestTemplate` does. Is that a possible replacement for this? `HttpClient` is immutable and therefore thread safe.

Reply:
Digging into the RestTemplate, I noticed this comment in the class:
NOTE: As of 5.0 this class is in maintenance mode, with only minor requests for changes and bugs to be accepted going forward. Please, consider using the org.springframework.web.reactive.client.WebClient which has a more modern API and supports sync, async, and streaming scenarios.

Should we consider a task to address this holistically?

Reply:
This appears to be resolved with the use of a spy instead of a mock.

Reply:
Great!  And also, yeah, we may want to migrate toe WebClient or the java native one